src/Microsoft.DotNet.Tools.Compiler.Native/ArgValues.cs

```diff
@@ -56,8 +57,20 @@ public NativeCompileSettings GetNativeCompileSettings()
 
             if (!string.IsNullOrEmpty(IlcPath))
             {
-                config.IlcPath = IlcPath;
-                config.IlcSdkPath = IlcPath;
+                // We want a directory path. If the user gave us the exact path to the executable
+                // then we can be helpful and convert that to the directory rather than forcing
+                // the command to be re-typed.
+                string ilcDir = IlcPath;
+                if(File.Exists(IlcPath) && !Directory.Exists(IlcPath))
+                {
+                    string potentialIlcDir = Path.GetDirectoryName(IlcPath);
+                    if(Directory.Exists(potentialIlcDir))
+                    {
+                        ilcDir = potentialIlcDir;
+                    }
+                }
+                config.IlcPath = ilcDir;
+                config.IlcSdkPath = ilcDir;
             }
```

Comment:
The "Path" term is overloaded. The option should be --ilcdir or --ilcexe. Anyway, LGTM.

Is the second DirectoryExists check necessary? (May be needed for unix, if somebody supplied: "/ilc.exe")

NIT: Just add a space between two `if(`